docs/src/main/asciidoc/security-customization.adoc

```diff
@@ -277,6 +277,30 @@ and add the latest BouncyCastle TLS dependency:
 </dependency>
 ----
 
+=== BouncyCastle FIPS
+
+If you need to register an `org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider` JCE provider then please set a `BCFIPS` provider name:
+
+.Example Security Providers BouncyCastle FIPS Configuration
+[source,properties]
+----
+quarkus.security.security-providers=BCFIPS
```

Comment:
IMHO this step is a bit cryptic (no pun intended) :)

It seems easier to do have something like `quarkus.security.fips.enabled=true` than to remember to use `BCFIPS` here

Reply:
@gastaldi users will have to remember either option or just check the docs :-)

`BCFIPS` is a well-known provider name known to anyone who wants to use Bouncy Castle and FIPS and it is consistent with how BouncyCastle itself recommends enabling BC + FIPS (i.e register a security provider named `BCFIPS`) and here we do the same, register this provider via `quarkus.security.security-providers`.

If we start with `quarkus.security.fips.enabled=true` then it will become ambiguous IMHO, should it work with the default `SunJSSE` which also has some FIPS code ? Or what does this mean, given the default providers section above:
```
quarkus.security.security-providers=SunRsaSign
quarkus.security.fips.enabled=true
```

So IMHO having an `atomic` `BCFIPS` is all right as we only plan to support FIPS in combination with BouncyCastle. Have I convinced you a little bit :-) ?

Reply:
Then `BCFIPSJSSE` or similar will follow, a single documented 'convention' name, BC docs recommend setting 3 providers, and some properties, but in Quarkus it will be a one liner - enable BC + FIPS to secure 2-way TLS

Reply:
Fair enough. I was under the impression that `quarkus.security.security-providers` would contain FQNs of `java.security.Provider` implementations

Reply:
@gastaldi see, we are not going to support `fips` generally, that is the problem with yet another property IMHO. We have `security-providers` one and that should be enough to register a provider who can do whatever it can do, it is generic